### ISSUE TITLE
Can't download 720p Big YouTube videos

### ISSUE THREAD

**Issue by noobimpostor2** [TeamNewPipe/NewPipe#8314]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.23.0

## Exception
* __User Action:__ download failed
* __Request:__ https://www.youtube.com/watch?v=8GE-FlT1fhc [ {type=video format=MPEG-4 quality=720p videoOnly=false} ]
* __Content Country:__ IN
* __Content Language:__ en-IN
* __App Language:__ en_IN
* __Service:__ YouTube
* __Version:__ 0.23.0
* __OS:__ Linux Android 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.net.ProtocolException: unexpected end of stream
	at com.android.okhttp.internal.http.Http1xStream$FixedLengthSource.read(Http1xStream.java:398)
	at com.android.okhttp.okio.RealBufferedSource$1.read(RealBufferedSource.java:372)
	at us.shandian.giga.get.DownloadRunnableFallback.run(DownloadRunnableFallback.java:98)
	at us.shandian.giga.get.DownloadRunnableFallback.run(DownloadRunnableFallback.java:131)
	at us.shandian.giga.get.DownloadRunnableFallback.run(DownloadRunnableFallback.java:131)
	at us.shandian.giga.get.DownloadRunnableFallback.run(DownloadRunnableFallback.java:131)

```
</details>
<hr>

**Comment by noobimpostor2** [comment]
Btw I know it's YouTube side issue , because since week I can't download any video with Newpipe , and also can't play on VLC player and also online downloader services not working on big YouTube videos.

I reported here because if anyone facing same issue or is just me?

**Comment by TiA4f8R** [comment]
YouTube has some (voluntary?) server issues with 720p video+audio (itag 22) on some videos.

I saw this issue recently on yt-dlp and youtube-dl issue trackers too. There is nothing we can do for this specific format right now, but we can switch to the video-only and audio merged variant (like we did for video playback to allow seamless transition when switching between other apps).

The issue seems to solve by itself when waiting some time (the video stream is again fetchable).
